src/test/javascript/spec/component/programming-exercise/tasks/programming-exercise-grading-tasks-table.component.spec.ts

```diff
@@ -0,0 +1,219 @@
+
+    const gradingStatistics = {} as ProgrammingExerciseGradingStatistics;
+    const exercise = { id: 1 } as ProgrammingExercise;
+    const course = { id: 2 } as Course;
+
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [ProgrammingExerciseGradingTasksTableComponent, MockPipe(ArtemisTranslatePipe), MockComponent(ButtonComponent)],
+            providers: [ProgrammingExerciseTaskService, { provide: TranslateService, useClass: MockTranslateService }],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ProgrammingExerciseGradingTasksTableComponent);
+                comp = fixture.componentInstance;
+
+                comp.gradingStatisticsObservable = of(gradingStatistics);
+                comp.exercise = exercise;
+                comp.course = course;
+                taskService = fixture.debugElement.injector.get(ProgrammingExerciseTaskService);
+                taskServiceUpdateTasksStub = jest.spyOn(taskService, 'updateTasks');
+            });
+    });
+
+    it('should pass parameters to the task service and configure it', () => {
+        const taskServiceConfigureStub = jest.spyOn(taskService, 'configure').mockReturnValue(of([]));
+
+        comp.ngOnInit();
+
+        expect(taskServiceConfigureStub).toHaveBeenCalledOnceWith(exercise, course, gradingStatistics);
+    });
+
+    it('should update tasks through the service and set them in the component', () => {
+        comp.tasks = [{ taskName: 'task1' }] as ProgrammingExerciseTask[];
+        const updatedTasks = [{ taskName: 'updatedTask' }] as ProgrammingExerciseTask[];
+        taskServiceUpdateTasksStub.mockReturnValue(updatedTasks);
+
+        comp.updateTasks();
+
+        expect(taskServiceUpdateTasksStub).toHaveBeenCalledOnce();
+        expect(comp.tasks).toBe(updatedTasks);
+    });
+
+    it('should pass inactive test case toggle to the task service and update the components tasks', () => {
+        comp.tasks = [{ taskName: 'task1' }] as ProgrammingExerciseTask[];
+        const inactiveTasks = [{ taskName: 'updatedTask' }] as ProgrammingExerciseTask[];
+        const taskServiceToggleIgnoreInactiveStub = jest.spyOn(taskService, 'toggleIgnoreInactive').mockReturnValue();
+        taskServiceUpdateTasksStub.mockReturnValue(inactiveTasks);
+
+        comp.toggleShowInactiveTestsShown();
+
+        expect(taskServiceToggleIgnoreInactiveStub).toHaveBeenCalledOnce();
+        expect(taskServiceUpdateTasksStub).toHaveBeenCalledOnce();
+    });
+
+    it('should pass saving to the service', () => {
+        const subject = new Subject();
+        const taskServiceSaveTestCasesStub = jest.spyOn(taskService, 'saveTestCases').mockReturnValue(subject as Observable<ProgrammingExerciseTask[]>);
+
+        comp.saveTestCases();
+
+        expect(comp.isSaving).toBeTruthy();
```

Comment:
I'd suggest ``toBeTrue()`` for the assertions on ``isSaving``. It's in any case a boolean, right?

Reply:
The added tests are not really part of this PR and only here because I renamed those files.

I will change this in a follow-up together with Dominiks suggestions